File: golang-code/ex8/empresa.go
package main

type Empresa struct {
	Nome       string
	Empregados []*Empregado
}

func (e *Empresa) AdicionarEmpregado(empregado *Empregado) {
	e.Empregados = append(e.Empregados, empregado)
}

func (e *Empresa) ListarEmpregados() {
	for _, empregado := range e.Empregados {
		println(empregado.String())
	}
}

File: java-code/src/main/java/ex8/Empresa.java
package ex8;

import java.util.ArrayList;
import java.util.List;

public class Empresa {
    private String nome;
    private List<Empregado> empregados;

    public Empresa(String nome) {
        this.nome = nome;
        this.empregados = new ArrayList<>();
    }

    public void adicionarEmpregado(Empregado empregado) {
        empregados.add(empregado);
    }


    public void listarEmpregados() {
        for (Empregado empregado : empregados) {
            System.out.println(empregado);
        }
    }

    public String getNome() {
        return nome;
    }
}
